File: lottoval/lottoval.py
class Game:
	def __init__(self, n, c, o):
		self.name = n
		self.cost = c
		self.odds = o
	def expected(self,prize):
		return prize * 1000000 / self.odds / self.cost

games = [Game("Powerball", 292201338.0, 2.0), 
	Game("Mega Millions", 302575350.0, 2.0),
	Game("Texas Lotto", 25827165.0, 1.0)]

print("Enter prizes in millions:")
for g in games:
	p = float(input(g.name + ": "))
	print(g.expected(p))
	print()

File: lottoval/lottoval.rb
class Game
	def initialize(name, odds, cost)
		@name = name
		@odds = odds
		@cost = cost
	end

	def expected(prize)
		prize * 1000000 / @odds / @cost
	end

	attr_reader :name
end

games = [Game.new("Powerball", 292201338.0, 2.0), 
	Game.new("Mega Millions", 302575350.0, 2.0),
	Game.new("Texas Lotto", 25827165.0, 1.0)]

puts "Enter game prizes in millions:"
games.each do |g|
    print "#{g.name}: "
    prize = gets.chop.to_f
    puts (g.expected prize) ; puts
end
